core/server/master/src/main/java/alluxio/master/file/contexts/OperationContext.java

```diff
@@ -11,54 +11,89 @@
 
 package alluxio.master.file.contexts;
 
+import com.google.protobuf.GeneratedMessageV3;
+
 import javax.annotation.concurrent.NotThreadSafe;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
 
 /**
  * Used as a base class for wrapping context around proto messages.
  *
  * @param <T> Proto message type
+ * @param <C> extended type
```

Comment:
Is the only reason for this, to have a `withTracker()` call? It seems inconvenient to add the same class as a template class parameter. Instead, can't we just have a `setTracker()`, that updates the instance, and we just pass in the instance to the rpc call?

Reply:
I'm just trying to make service-handler clean by making the API pretty to use.